packages/ember-truth-helpers/src/helpers/or.ts

```diff
@@ -5,20 +5,20 @@ interface OrSignature<T extends unknown[]> {
   Args: {
     Positional: T;
   };
-  Return: T[number];
+  Return: NonNullable<T[number]>;
```

Comment:
I think this probably covers the majority of uses cases, like the one I highlighted here
https://github.com/jmurphyau/ember-truth-helpers/pull/193#discussion_r1314385402

Though there's a few problems like...
- technically it doesn't cover the first arg if it were other falsey values like `false`, `0`.... etc
- what if all args are nullable? `@mustBeDefined={{or undefined undefined}}` wouldn't cause any types issues

Are we able to use the same types that were defined in the gavant package? Was there a problem with them?